File: frontend/src/types/player.ts
export type PlayerStats = {
  str: number
  dex: number
  con: number
  int: number
  wis: number
  cha: number
}

export type Player = {
  id: string
  game_id: string
  profile_id: string
  character_name: string
  character_class: string
  hp_current: number
  hp_max: number
  stats: PlayerStats
  status: string
  joined_at: string
}

export type InventoryItem = {
  id: string
  player_id: string
  item_name: string
  quantity: number
  properties: Record<string, unknown> | null
  created_at: string
}

export type CharacterData = {
  id: string
  character_name: string
  character_class: string
  hp_current: number
  hp_max: number
  stats: Record<string, number>
}

File: backend/models/player.py
"""Player domain model."""

from pydantic import BaseModel
from typing import Any


class Player(BaseModel):
    """Represents a player in a D&D game."""

    id: str
    game_id: str
    profile_id: str
    character_name: str
    character_class: str
    race: str = "Human"
    level: int = 1
    hp_current: int
    hp_max: int
    stats: dict[str, Any]
    status: str
